lib/xray_creator.py

```diff
@@ -47,10 +47,7 @@ def _initialize_books(self):
         book_lookup = {}
         for book in self.books_not_failing():
             self._total_not_failing += 1
-            if book_lookup.has_key(book.title):
-                book_lookup[book.title].append(book)
-                continue
-            book_lookup[book.title] = [book] 
+            book_lookup[self._db.field_for('uuid', book.book_id)] = book
```

Comment:
AFAIK UUID should be unique (I assume that is one of the "U"), but it wouldn't hurt to add an `assert not self._db.field_for('uuid', book.book_id) in book_lookup`.  Probably storing `self._db.field_for('uuid', book.book_id)` in a variable for clarity/possibly speed.